automatic_data_generation/train_and_eval_cvae.py

```diff
@@ -64,38 +64,44 @@ def train_and_eval_cvae(args):
     if args.conditioning == NO_CONDITIONING:
         args.conditioning = None
 
-    model = CVAE(
-        conditional=args.conditioning,
-        compute_bow=args.bow_loss,
-        vocab_size=dataset.vocab_size,
-        embedding_size=args.embedding_dimension,
-        rnn_type=args.rnn_type,
-        hidden_size=args.hidden_size,
-        word_dropout_rate=args.word_dropout_rate,
-        embedding_dropout_rate=args.embedding_dropout_rate,
-        z_size=args.latent_size,
-        n_classes=dataset.n_classes,
-        sos_idx=dataset.sos_idx,
-        eos_idx=dataset.eos_idx,
-        pad_idx=dataset.pad_idx,
-        unk_idx=dataset.unk_idx,
-        max_sequence_length=args.max_sequence_length,
-        num_layers=args.num_layers,
-        bidirectional=args.bidirectional,
-        temperature=args.temperature,
-        force_cpu=args.force_cpu
-    )
-    if args.load_folder:
-        model.from_folder(args.load_folder)
-        LOGGER.info('Loaded model from %s' % args.load_folder)
-
+    if not args.load_folder:
+        model = CVAE(
+            conditional=args.conditioning,
+            compute_bow=args.bow_loss,
+            vocab_size=dataset.vocab_size,
+            embedding_size=args.embedding_dimension,
+            rnn_type=args.rnn_type,
+            hidden_size=args.hidden_size,
+            word_dropout_rate=args.word_dropout_rate,
+            embedding_dropout_rate=args.embedding_dropout_rate,
+            z_size=args.latent_size,
+            n_classes=dataset.n_classes,
+            cat_size=dataset.n_classes if args.cat_size is None else args.cat_size, 
+            sos_idx=dataset.sos_idx,
+            eos_idx=dataset.eos_idx,
+            pad_idx=dataset.pad_idx,
+            unk_idx=dataset.unk_idx,
+            max_sequence_length=args.max_sequence_length,
+            num_layers=args.num_layers,
+            bidirectional=args.bidirectional,
+            temperature=args.temperature,
+            force_cpu=args.force_cpu
+        )
+    else:
+        model = CVAE.from_folder(args.load_folder)
+        LOGGER.info('Loaded model from %s' % args.load_folder)                                                
+        
+    vectors = dataset.text.vocab.vectors if args.input_type=='utterance' else dataset.delex.vocab.vectors
+    model.load_embedding(vectors)
```

Comment:
Why not do so in the __init__ of the model class? Would necessitate to make vectors an attribute of Dataset and pass it as an argument to the model